MapboxAndroidDemo/build.gradle

```diff
@@ -10,6 +10,11 @@ android {
     compileSdkVersion androidVersions.compileSdkVersion
     buildToolsVersion androidVersions.buildToolsVersion
 
+    compileOptions {
```

Comment:
I don't think you need this `compileOptions`section anymore. I tried removing it and the example works just fine for me.